Mage.Server/src/main/java/mage/server/UserManager.java

```diff
@@ -231,8 +231,8 @@ private void checkExpired() {
             }
             logger.debug("Users to remove " + toRemove.size());
             final Lock w = lock.readLock();
-            w.lock();
             try {
+                w.lock();
```

Comment:
Same... `w.lock();` must be outside from try-finally block... [look at discussion from previous pull request](https://github.com/magefree/mage/pull/4648/files#r176650560)